File: selectionsort/selectionsort.js
function selectionSort() {

    var array = [2, 7, 4, 1, 5, 3];
    document.getElementById("array").value = array.toString();

    var len = array.length;

    for (var i = 0; i < len - 1; i++) {
        var minIndex = i;
        for (var j = i + 1; j < len; j++) {

            if (array[j] < array[minIndex]) {
                minIndex = j;
            }
        }
        var temp = array[i];
        array[i] = array[minIndex];
        array[minIndex] = temp;
    }
    document.getElementById("sortedarray").value = array.toString();
}

File: selectionsort/SelectionSort.java
package org.com.sort;

import java.util.Arrays;

public class SelectionSort {
	public static void sort(int[] a){
		int sorted[]=new int[a.length];
		for(int i=0;i<a.length-1;i++){
			int[] minindex=minimum(a,i);
			a[minindex[1]]=a[i];
			a[i]=minindex[0];
			sorted[i]=minindex[0];
		}
		printArray(a);
	}
	
	public static int[] minimum(int[] a, int j){
		int min=a[j];
		int index=0;
		for(int i=j+1;i<a.length;i++){
			if(min>a[i]){
				min=a[i];
				index=i;
			}
		}
		return new int[]{min,index};
	}
	public static void printArray(int[] sorted){
		System.out.println("Selection sort: "+ Arrays.toString(sorted));
	}
	public static void main(String args[]){
		int a[] = {10,13,7,8,6};
		SelectionSort.sort(a);
	}
	
}
